Event Planning and Collaboration Tool/src/main/java/bg/sofia/uni/fmi/web/project/service/ContractService.java

```diff
@@ -0,0 +1,153 @@
+package bg.sofia.uni.fmi.web.project.service;
+
+import bg.sofia.uni.fmi.web.project.dto.ContractDto;
+import bg.sofia.uni.fmi.web.project.model.Contract;
+import bg.sofia.uni.fmi.web.project.repository.ContractRepository;
+import bg.sofia.uni.fmi.web.project.validation.MethodNotAllowed;
+import bg.sofia.uni.fmi.web.project.validation.ResourceNotFoundException;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@Validated
+@AllArgsConstructor
+public class ContractService {
+    private final ContractRepository contractRepository;
+
+    public long addContract(@NotNull(message = "The given vendor cannot be null!")
+                            Contract contractToSave) {
+        Contract contract = contractRepository.save(contractToSave);
+        checkForSaveException(contract);
```

Comment:
I think that the method checkForSaveException() is not strictly necessary since the save method is expected to throw an exception if there are any problems which will result in the stack trace.

Reply:
Yes, but nobody will understand anything of the stack trace and as an addition to that the main reason behind this check is to display a custom brief error message so as to show that there was problem with the repository, either from a connection, or some other reason .

Reply:
Yes, I see the point. The concern is that if there is a problem in the save method, some sort of exception will be thrown and your method checkForSaveException will never be reached as the exception won't be caught. If you want to make a custom brief error message in the form of a custom exception, then you have to use try-catch block around the save method instead of checking the result of the save method. If there is an exception with the connection, the result of the save method which you store in the "contract" variable won't be null. It wont't be assigned at all.

Reply:
I got your point and it will be fixed